frame/indices/src/lib.rs

```diff
@@ -82,36 +74,145 @@ decl_event!(
 		<T as frame_system::Trait>::AccountId,
 		<T as Trait>::AccountIndex
 	{
-		/// A new account index was assigned.
-		///
-		/// This event is not triggered when an existing index is reassigned
-		/// to another `AccountId`.
-		NewAccountIndex(AccountId, AccountIndex),
+		/// A account index was assigned.
+		IndexAssigned(AccountId, AccountIndex),
+		/// A account index has been freed up (unassigned).
+		IndexFreed(AccountIndex),
 	}
 );
 
-decl_storage! {
-	trait Store for Module<T: Trait> as Indices {
-		/// The next free enumeration set.
-		pub NextEnumSet get(fn next_enum_set) build(|config: &GenesisConfig<T>| {
-			(config.ids.len() as u32 / ENUM_SET_SIZE).into()
-		}): T::AccountIndex;
-
-		/// The enumeration sets.
-		pub EnumSet get(fn enum_set) build(|config: &GenesisConfig<T>| {
-			(0..((config.ids.len() as u32) + ENUM_SET_SIZE - 1) / ENUM_SET_SIZE)
-				.map(|i| (
-					i.into(),
-					config.ids[
-						(i * ENUM_SET_SIZE) as usize..
-						config.ids.len().min(((i + 1) * ENUM_SET_SIZE) as usize)
-					].to_owned(),
-				))
-				.collect::<Vec<_>>()
-		}): map hasher(blake2_256) T::AccountIndex => Vec<T::AccountId>;
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		/// The index was not already assigned.
+		NotAssigned,
+		/// The index is assigned to another account.
+		NotOwner,
+		/// The index was not available.
+		InUse,
+		/// The source and destination accounts are identical.
+		NotTransfer,
 	}
-	add_extra_genesis {
-		config(ids): Vec<T::AccountId>;
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin, system = frame_system {
+		fn deposit_event() = default;
+
+		fn on_initialize() {
+			Self::migrations();
+		}
+
+		/// Assign an previously unassigned index.
+		///
+		/// Payment: `Deposit` is reserved from the sender account.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// - `index`: the index to be claimed. This must not be in use.
+		///
+		/// Emits `IndexAssigned` if successful.
+		///
+		/// # <weight>
+		/// - `O(1)`.
+		/// - One storage mutation (codec `O(1)`).
+		/// - One reserve operation.
+		/// - One event.
+		/// # </weight>
+		fn claim(origin, index: T::AccountIndex) {
```

Comment:
This function allows a single account to claim multiple indices. Is that intended?

Reply:
yes.